packages/agreement-process/src/services/agreementService.ts

```diff
@@ -192,66 +277,162 @@ export function agreementServiceBuilder(
       { authData, correlationId, logger }: WithLogger<AppContext>
     ): Promise<void> {
       logger.info(`Deleting agreement ${agreementId}`);
-      const agreement = await agreementQuery.getAgreementById(agreementId);
+      const agreement = await retrieveAgreement(agreementId, agreementQuery);
+
+      assertRequesterIsConsumer(agreement.data, authData);
+
+      assertExpectedState(
+        agreementId,
+        agreement.data.state,
+        agreementDeletableStates
+      );
+
+      for (const d of agreement.data.consumerDocuments) {
+        await fileManager.delete(config.s3Bucket, d.path, logger);
+      }
 
       await repository.createEvent(
-        await deleteAgreementLogic(
-          {
-            agreementId,
-            authData,
-            deleteFile: fileManager.delete,
-            agreement,
-          },
-          correlationId,
-          logger
+        toCreateEventAgreementDeleted(
+          agreement.data,
+          agreement.metadata.version,
+          correlationId
         )
       );
     },
     async submitAgreement(
       agreementId: AgreementId,
       payload: ApiAgreementSubmissionPayload,
-      ctx: WithLogger<AppContext>
+      { authData, correlationId, logger }: WithLogger<AppContext>
     ): Promise<Agreement> {
-      ctx.logger.info(`Submitting agreement ${agreementId}`);
-      const [agreement, updatesEvents] = await submitAgreementLogic(
-        agreementId,
-        payload,
-        contractBuilder(
-          ctx.authData.selfcareId,
-          attributeQuery,
-          fileManager.storeBytes,
-          ctx.logger
-        ),
-        eserviceQuery,
-        agreementQuery,
-        tenantQuery,
-        ctx
+      logger.info(`Submitting agreement ${agreementId}`);
+
+      const agreement = await retrieveAgreement(agreementId, agreementQuery);
+
+      assertRequesterIsConsumer(agreement.data, authData);
+      assertSubmittableState(agreement.data.state, agreement.data.id);
+      await verifySubmissionConflictingAgreements(
+        agreement.data,
+        agreementQuery
       );
 
+      const [agreementdocumentSeed, updatesEvents] =
+        await processSubmitAgreement({
+          agreementData: agreement,
+          eservice: await retrieveEService(
+            agreement.data.eserviceId,
+            eserviceQuery
+          ),
+          payload,
+          agreementQuery,
+          tenantQuery,
+          constractBuilder: contractBuilder(
```

Comment:
Typo, plus I would suggest the same here as well: instance it in a dedicated const above then pass it as param 

```suggestion
          contractBuilder: contractBuilder(
```

Reply:
I prefer to not create a const for this since `contractBuilder` is already a const